Project/Wikode/wikodeApp/utils/activityManager.py

```diff
@@ -0,0 +1,334 @@
+            "published": self.getCurrentTimeAsISO(),
+            "actor": {
+                "type": "Person",
+                "id": self.getOwnerURL(),
+                "name": self.getOwnerName(),
+                "url": self.getOwnerURL()
+            },
+            "object": {
+                "id": activity_target_url,
+                "type": activity_target_type,
+                "url": activity_target_url,
+                "name": activity_target_name
+            }
+        }
+
+        activity = Activity(
+            user_id=self.user_id,
+            activity_type=5,
+            target_type=2,
+            target_id=target_id,
+            activity_JSON=json
+        )
+        activity.save()
+
+    # This method saves a tag activity for a whole article
+    # target_id: the id of the article which is being tagged
+    # tag_id: the id of the tag which is being used for tagging
+    def saveTaggingActivityForArticle(self, target_id, tag_id):
+        target = self.getTargetAsArticle(target_id)
+        if isinstance(target, Article):
+            activity_target_type = 'Article'
+            activity_target_url = self.getTagURL(id=target_id)
+            activity_target_name = target.Title
+
+        json = {
+            "@context": "https://www.w3.org/ns/activitystreams",
+            "summary": "{} tagged {}".format(self.getOwnerName(), activity_target_name),
+            "type": "Add",
+            "published": self.getCurrentTimeAsISO(),
+            "actor": {
+                "type": "Person",
+                "id": self.getOwnerURL(),
+                "name": self.getOwnerName(),
+                "url": self.getOwnerURL()
+            },
+            "tag": {
+                 "id": self.getTagURL(tag_id)
+             },
+            "object": {
+                "id": activity_target_url,
+                "type": activity_target_type,
+                "url": activity_target_url,
+                "name": activity_target_name
+            }
+        }
+
+        activity = Activity(
+            user_id=self.user_id,
+            activity_type=6,
+            target_type=3,
+            target_id=target_id,
+            activity_JSON=json
+        )
+        activity.save()
+
+    # This method saves an annotation activity
+    # target_article_id: The id of the article, which has the fragment
+    # tag_id: the id of the tag, which is being used for tagging
+    # start_index: the beginning index of fragment inside the text
+    # end_index: the finishing index of fragment inside the text
+    def saveAnnotationActivity(self, target_article_id, tag_id, start_index, end_index):
+        tag_object = self.getTargetAsTag(target_id=tag_id)
+        json = {
+                 "@context": "http://www.w3.org/ns/anno.jsonld",
+                 "id": self.getTagURL(id=tag_id),
+                 "type": "Annotation",
+                "body": [
+                     {
+                      "type": "TextualBody",
```

Comment:
Burada tag body'si olarak wiki URL'leri kullanmak istebiliriz takımla bir konuşalım.

Reply:
body içine tag'in Q code'u da eklenecek.